From 4f19230663e5fa645e40df0e519273dde5c8df36 Mon Sep 17 00:00:00 2001
From: Axel Davy <axel.davy@ens.fr>
Date: Wed, 26 Nov 2014 12:30:35 +0100
Subject: [PATCH 16/21] d3dadapter: implement
 DRI3PresentGroup_CreateAdditionalPresent

The implementation is probably not optimal, there are the following questions:

. what if two swapchain render to the same X drawable (even if different Win HWND)
Current implementation is going to behave very badly there
. spec is not precise here: on which screen should we be on ? Since we don't support
multiple screen/displays yet, this is not yet pb.
. What about focus window ?

Signed-off-by: Axel Davy <axel.davy@ens.fr>
---
 dlls/winex11.drv/d3dadapter.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/d3dadapter.c b/dlls/winex11.drv/d3dadapter.c
index 5ea876b..c8624c5 100644
--- a/dlls/winex11.drv/d3dadapter.c
+++ b/dlls/winex11.drv/d3dadapter.c
@@ -713,8 +713,10 @@ DRI3PresentGroup_CreateAdditionalPresent( struct DRI3PresentGroup *This,
                                           D3DPRESENT_PARAMETERS *pPresentationParameters,
                                           ID3DPresent **ppPresent )
 {
-    FIXME("(%p, %p, %p), stub!\n", This, pPresentationParameters, ppPresent);
-    return D3DERR_INVALIDCALL;
+    HRESULT hr;
+    hr = DRI3Present_new(gdi_display, This->present_backends[0]->devname,
+                         pPresentationParameters, 0, (struct DRI3Present **)ppPresent);
+    return hr;
 }
 
 static void WINAPI
-- 
2.2.2

